eZ/Publish/Core/Persistence/Legacy/User/Role/Gateway/DoctrineDatabase.php

```diff
@@ -47,15 +47,21 @@ public function __construct(DatabaseHandler $handler)
      */
     public function createRole(Role $role)
     {
-        // Role ID is set when creating a draft from an existing role
+        // Role original ID is set when creating a draft from an existing role
+        if ($role->status === Role::STATUS_DRAFT && $role->id) {
+            $roleOriginalId = $role->id;
+        } elseif ($role->status === Role::STATUS_DRAFT) {
+            $roleOriginalId = -1;
+        } else {
+            $roleOriginalId = Role::STATUS_DEFINED;
```

Comment:
Alternative, just add inline comment there making it clear Role::STATUS_DEFINED equals 0, so it is clear from reading these lines how db values will look like.